pkg/policies/policies.go

```diff
@@ -637,10 +674,12 @@ func LoadPolicyScriptsFromSpec(policy *v1.Policy, kind v1.CraftingSchema_Materia
 					return nil, fmt.Errorf("failed to load policy script: %w", err)
 				}
 
-				// Inject boilerplate if needed
-				script, err = rego.InjectBoilerplate(script, policy.GetMetadata().GetName())
-				if err != nil {
-					return nil, fmt.Errorf("failed to inject boilerplate: %w", err)
+				// Inject boilerplate only for Rego policies, not WASM
+				if engine.DetectPolicyType(script) == engine.PolicyTypeRego {
```

Comment:
I wonder if the policy type should be a property of `spec`. I'm ok if we feel comfortable with this magic detection, but this might not stand in the future if we support more engines (`jq` or something else), since here ware just detecting that this is NOT WASM, but it might be anything else.